machine/v1beta1/types_awsprovider.go

```diff
@@ -95,6 +95,16 @@ type AWSMachineProviderConfig struct {
 	// The field size should be greater than 0 and the field input must start with cr-***
 	// +optional
 	CapacityReservationID string `json:"capacityReservationId"`
+	// marketType specifies the type of market for the EC2 instance.
+	// Valid values are OnDemand, Spot, CapacityBlock and omitted.
+	// Defaults to OnDemand.
+	// When set to OnDemand the instance runs as a standard OnDemand instance.
+	// When SpotMarketOptions is provided, the marketType defaults to "Spot".
```

Comment:
I would move this up a line so that everything related to defaulting behavior is grouped together.

Additionally, white space can be helpful to denote logical blocks of information on comments. It might be nice to add some spacing throughout the Godoc here to do just that.